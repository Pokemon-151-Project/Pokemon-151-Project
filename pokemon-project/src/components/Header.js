import React from "react";

import Dropdown from "./Dropdown";

//I'd like to style this better. Maybe put h1 on the left, display options in middle, then my info and links on the right.
const Header = (props) => {
  const { changeDisplay } = props;
  return (
    <header>
      <h1>Pokemon Display</h1>

      {/* Begin display options dropdown. See Dropdown.js */}
      <Dropdown changeDisplay={changeDisplay} />

      {/* section for info about me */}
      <div className="adam-info">
        <h2 className="header-item">Author: Adam Hinton</h2>

        <div className="header-links header-item">
          <a
            href="https://github.com/adamhinton/pokemon-project"
            target="_blank"
            rel="noreferrer"
          >
            Source
          </a>

          <a
            href="https://bulbapedia.bulbagarden.net"
            target="_blank"
            rel="noreferrer"
          >
            Bulbapedia
          </a>

          <a href="https://pokeapi.co" target="_blank" rel="noreferrer">
            PokeAPI
          </a>
        </div>
      </div>
    </header>
  );
};

export default Header;
