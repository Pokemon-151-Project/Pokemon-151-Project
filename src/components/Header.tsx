import React from "react";
import styled from "styled-components";
import Dropdown from "./Dropdown";
import useDarkMode from "../hooks/useDarkMode";
import { BsMoon, BsSun } from "react-icons/bs";
import linksArray from "../utils/Headerutils/linksArray";
import linkMaker from "../utils/Headerutils/linkMaker";
import { PokemonTypes } from "../data";

const HeaderContainer = styled.header`
	display: flex;
	flex-direction: column;
	align-items: center;
	background-color: #f2f2f2;
`;

const DarkModeSection = styled.div`
	display: flex;
	align-items: center;
`;

const ShinyButtonDiv = styled.div`
	margin-left: 20px;
`;

const ShinyButton = styled.button`
	background-color: #4caf50;
	color: white;
	padding: 14px 20px;
	margin: 8px 0;
	border: none;
	cursor: pointer;
	width: 100%;
`;

const AdamInfo = styled.section`
	text-align: center;
`;

interface Props {
	changeDisplay: (
		type?: PokemonTypes | null,
		size?: string | null,
		height?: string | null,
		reset?: string | null
	) => void;
	setIsShiny: React.Dispatch<React.SetStateAction<boolean>>;
	isShiny: boolean;
	toggleTheme: () => void;
}

const Header: React.FC<Props> = ({
	changeDisplay,
	setIsShiny,
	isShiny,
	toggleTheme,
}) => {
	const [isDarkMode, setDarkMode] = useDarkMode();

	return (
		<HeaderContainer>
			<h1>Pokemon 151 Project</h1>
			<section>
				<menu>
					<DarkModeSection>
						<div data-testid="logo">
							{isDarkMode ? "Dark Mode" : "Light Mode"}
						</div>
						<button
							data-testid="toggle_btn"
							onClick={() => {
								setDarkMode(!isDarkMode);
							}}
						>
							{isDarkMode ? (
								<BsSun color="#ff0" size="24" title="Switch to light mode" />
							) : (
								<BsMoon size="24" title="Switch to dark mode" />
							)}
						</button>
					</DarkModeSection>
					<ShinyButtonDiv>
						<ShinyButton
							onClick={() => {
								setIsShiny(!isShiny);
							}}
						>
							{isShiny ? "Show Non-Shiny Forms" : "Show Shiny Forms"}
						</ShinyButton>
					</ShinyButtonDiv>
				</menu>
				<Dropdown changeDisplay={changeDisplay} />
			</section>
			<AdamInfo>
				<h2>Author: Adam Hinton</h2>
				<nav>{linkMaker(linksArray)}</nav>
			</AdamInfo>
		</HeaderContainer>
	);
};

export default Header;
