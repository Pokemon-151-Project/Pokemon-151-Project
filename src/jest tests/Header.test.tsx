import Header from "../components/Header";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import React from "react";

const mockChangeDisplay = jest.fn();
const mockSetIsShiny = jest.fn();
const mockToggleTheme = jest.fn(); // Add this line

test("[1] Component renders without errors", () => {
	render(
		<Header
			changeDisplay={mockChangeDisplay}
			isShiny={false}
			setIsShiny={mockSetIsShiny}
			toggleTheme={mockToggleTheme}
			isDarkMode={true}
		/>
	);
});

test("[2] Renders the expected segments of the header", () => {
	render(
		<Header
			changeDisplay={mockChangeDisplay}
			isShiny={false}
			setIsShiny={mockSetIsShiny}
			toggleTheme={mockToggleTheme}
			isDarkMode={true}
		/>
	);
	const h1 = screen.queryByText(/pokemon 151 project/i);
	const author = screen.queryByText(/author: adam hinton/i);
	const source = screen.queryByText(/author/i);
	const dropbtn = screen.queryByText(/display options/i);

	expect(author).toBeInTheDocument();
	expect(h1).toBeInTheDocument();
	expect(source).toBeInTheDocument();
	expect(dropbtn).toBeVisible();
});

test("[3] 'Dark Mode' text changes to 'Light Mode' and vice versa when you click dark mode toggle logo", async () => {
	render(
		<Header
			changeDisplay={mockChangeDisplay}
			isShiny={false}
			setIsShiny={mockSetIsShiny}
			toggleTheme={mockToggleTheme}
			isDarkMode={true}
		/>
	);
	const darkModeToggleLogo = screen.getByTestId("toggle_btn");

	expect(screen.queryByText(/dark mode/i)).not.toBeVisible();
	expect(screen.queryByText(/light mode/i)).toBeVisible();

	await userEvent.click(darkModeToggleLogo);

	expect(screen.queryByText(/dark mode/i)).toBeVisible();
	expect(screen.queryByText(/light mode/i)).not.toBeVisible();
});

test("[4] Matches snapshot so nothing changes inadvertently", () => {
	const component = renderer.create(
		<Header
			changeDisplay={mockChangeDisplay}
			isShiny={false}
			setIsShiny={mockSetIsShiny}
			toggleTheme={mockToggleTheme}
			isDarkMode={true}
		/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
