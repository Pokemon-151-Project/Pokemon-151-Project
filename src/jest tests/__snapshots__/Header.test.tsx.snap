// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[4] Matches snapshot so nothing changes inadvertently 1`] = `
<header
  className="sc-jTQDJr eaKBoq"
>
  <h1
    className="sc-gLLuHO kwZjLc"
  >
    Pokemon 151 Project
  </h1>
  <section
    className="sc-iBdoyZ bLAImq"
  >
    <menu
      className="sc-fsYfxw kxxrzu"
    >
      <div
        className="sc-qZruQ eYmnhz"
        data-testid="logo"
      >
         
        Dark Mode
      </div>
      <button
        className="sc-jsEeA-d lmpeGT"
        data-testid="toggle_btn"
        onClick={[Function]}
      >
        <svg
          color="#ff0"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#ff0",
            }
          }
          viewBox="0 0 16 16"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Switch to light mode
          </title>
          <path
            d="M8 11a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0 1a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
          />
        </svg>
      </button>
      <div
        className="sc-irLvoH Vvvli"
      />
      <button
        className="sc-kFCrIq kIpQeD"
        onClick={[Function]}
      >
        Show Shiny Forms
      </button>
    </menu>
    <section
      className="sc-guDMob kkcneg"
    >
      <button
        className="sc-dmyDmy bbdhUi"
        data-testid="dropbtn"
      >
        Display Options
      </button>
      <menu
        className="sc-beyTiQ cvkhIn dropdown-content"
      >
        <div>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="reset-btn"
            id="reset-btn"
            onClick={[Function]}
          >
            Reset All
          </button>
          <h5
            className="sc-eDLKEg cQxIQw"
            data-testid="dropdown-display-title"
          >
            Pick a type:
          </h5>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="normal-btn"
            onClick={[Function]}
          >
            Normal
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="fire-btn"
            onClick={[Function]}
          >
            Fire
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="water-btn"
            onClick={[Function]}
          >
            Water
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="electric-btn"
            onClick={[Function]}
          >
            Electric
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="grass-btn"
            onClick={[Function]}
          >
            Grass
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="ice-btn"
            onClick={[Function]}
          >
            Ice
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="fighting-btn"
            onClick={[Function]}
          >
            Fighting
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="poison-btn"
            onClick={[Function]}
          >
            Poison
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="ground-btn"
            onClick={[Function]}
          >
            Ground
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="flying-btn"
            onClick={[Function]}
          >
            Flying
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="psychic-btn"
            onClick={[Function]}
          >
            Psychic
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="bug-btn"
            onClick={[Function]}
          >
            Bug
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="rock-btn"
            onClick={[Function]}
          >
            Rock
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="ghost-btn"
            onClick={[Function]}
          >
            Ghost
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="dragon-btn"
            onClick={[Function]}
          >
            Dragon
          </button>
        </div>
        <div>
          <h5
            className="sc-eDLKEg cQxIQw"
          >
            Pick a weight:
          </h5>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="small-btn"
            onClick={[Function]}
          >
            Small
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="big-btn"
            onClick={[Function]}
          >
            Big
          </button>
        </div>
        <div>
          <h5
            className="sc-eDLKEg cQxIQw"
          >
            Pick a height:
          </h5>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="short-btn"
            onClick={[Function]}
          >
            Short
          </button>
          <button
            className="sc-hLQTFJ piuou"
            data-testid="dropbtn-inner"
            id="tall-btn"
            onClick={[Function]}
          >
            Tall
          </button>
        </div>
      </menu>
    </section>
  </section>
  <section
    className="sc-csKJRI fDtcLf"
  >
    <h2>
      Author: Adam Hinton
    </h2>
    <nav
      className="sc-eTNxZ hHxGhQ"
    >
      <a
        className="sc-brSaGm fJNlLG"
        href="https://github.com/adamhinton/pokemon-project"
      >
        Source
      </a>
      <a
        className="sc-brSaGm fJNlLG"
        href="https://www.linkedin.com/in/adam-hinton/"
      >
        LinkedIn
      </a>
      <a
        className="sc-brSaGm fJNlLG"
        href="https://pokeapi.co"
      >
        PokeAPI
      </a>
      <a
        className="sc-brSaGm fJNlLG"
        href="https://bulbapedia.bulbagarden.net/wiki/Shiny_Pok%C3%A9mon"
      >
        Shinies
      </a>
    </nav>
  </section>
</header>
`;
